docs/installation.rst

```diff
@@ -1,11 +1,46 @@
 Installation
 ============
 
-To install this package run
+To use StarkNet.py, ``ecdsa, fastecdsa, sympy`` dependencies are required. Depending on the operating system,
+different install steps must be performed.
 
-``pip install starknet.py``
+Linux
+^^^^^
 
-or using Poetry:
+.. code-block:: console
 
-``poetry add starknet.py``
+    sudo apt install -y libgmp3-dev
+    pip install starknet-py
 
+MacOS with Intel processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+
+.. code-block:: console
+
+    brew install gmp
+    pip install starknet-py
+
+MacOS with Apple silicon
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+First install gmp:
+
+.. code-block:: console
+
+    brew install gmp
+
+Then install required dependencies and StarkNet.py itself:
+
+.. code-block:: console
+
+    CFLAGS=-I`brew --prefix gmp`/include LDFLAGS=-L`brew --prefix gmp`/lib pip install ecdsa fastecdsa sympy
```

Comment:
I think we can just do pip install starknet-py here. This way after uninstalling starknet-py dependencies won't be left in the system.

Reply:
Does it work if you just do `CFLAGS=-I`brew --prefix gmp`/include LDFLAGS=-L`brew --prefix gmp`/lib pip install starknet-py`? Personally I always did fastecdsa and the rest first, then starknet.py

Reply:
Yes it does, dependencies are installed with starknet.py and proper env variables are available.